rpc_state_reader_sn_api/src/lib.rs

```diff
@@ -0,0 +1,994 @@
+                stark_felt!("0"),
+                stark_felt!("9"),
+                stark_felt!("9"),
+                stark_felt!("4"),
+                stark_felt!("4254432d55534443"),
+                stark_felt!("f02e7324ecbd65ce267"),
+                stark_felt!("5754492d55534443"),
+                stark_felt!("8e13050d06d8f514c"),
+                stark_felt!("4554482d55534443"),
+                stark_felt!("f0e4a142c3551c149d"),
+                stark_felt!("4a50592d55534443"),
+                stark_felt!("38bd34c31a0a5c"),
+            ])
+        );
+        assert_eq!(
+            tx_trace.function_invocation.retdata,
+            Some(vec![0u128.into()])
+        );
+        assert_eq!(
+            tx_trace.function_invocation.execution_resources,
+            VmExecutionResources {
+                n_steps: 2808,
+                n_memory_holes: 136,
+                builtin_instance_counter: HashMap::from([
+                    ("range_check_builtin".to_string(), 49),
+                    ("pedersen_builtin".to_string(), 14),
+                ]),
+            }
+        );
+        assert_eq!(tx_trace.function_invocation.internal_calls.len(), 1);
+        assert_eq!(
+            tx_trace.function_invocation.internal_calls[0]
+                .internal_calls
+                .len(),
+            1
+        );
+        assert_eq!(
+            tx_trace.function_invocation.internal_calls[0].internal_calls[0]
+                .internal_calls
+                .len(),
+            7
+        );
+
+        assert_eq!(
+            tx_trace.fee_transfer_invocation.calldata,
+            Some(vec![
+                stark_felt!("1176a1bd84444c89232ec27754698e5d2e7e1a7f1539f12027f28b23ec9f3d8"),
+                stark_felt!("2b0322a23ba4"),
+                stark_felt!("0"),
+            ])
+        );
+        assert_eq!(
+            tx_trace.fee_transfer_invocation.retdata,
+            Some(vec![1u128.into()])
+        );
+        assert_eq!(
+            tx_trace.fee_transfer_invocation.execution_resources,
+            VmExecutionResources {
+                n_steps: 586,
+                n_memory_holes: 42,
+                builtin_instance_counter: HashMap::from([
+                    ("range_check_builtin".to_string(), 21),
+                    ("pedersen_builtin".to_string(), 4),
+                ]),
+            }
+        );
+        assert_eq!(tx_trace.fee_transfer_invocation.internal_calls.len(), 1);
+    }
+
+    #[test]
+    fn test_get_transaction_receipt() {
+        let rpc_state = RpcState::new(
+            RpcChain::MainNet,
+            BlockValue::Tag(serde_json::to_value("latest").unwrap()),
+        );
+        let tx_hash = TransactionHash(stark_felt!(
+            "06da92cfbdceac5e5e94a1f40772d6c79d34f011815606742658559ec77b6955"
+        ));
+
+        rpc_state.get_transaction_receipt(&tx_hash);
+    }
+}
+
+mod blockifier_transaction_tests {
```

Comment:
Yep, we should do a big refactor because there are too many types crammed in a single file, but as I said the priority now is not to make it pretty (if we keep extending the funcionality we may deem it necessary). The last module will be fidgeted with a lot when debugging.